### DISCUSSION TITLE
Using envvars in extend

### DISCUSSION THREAD

**Discussion by jayvdb** [sagiegurari/cargo-make#1163] [General]
Would it be possible to use envvars when using `extend`

```toml
extend = [{ path = "${CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY}/makefiles/tools.toml" }]

**Comment by sagiegurari** [comment]
yes but that would use env vars already defined and not those set by cargo make env blocks.
please open an issue so i can track it

**Reply by sagiegurari** [reply]
wait, i see your example uses env that caego make sets. so obviously no.

**Comment by jayvdb** [comment]
Is there another way to have Makefile.toml at multiple levels, and makefile tasks defined in a common file at the top of the repo?  i.e. without having various levels of "../../../" in the members Makefile.toml

**Reply by sagiegurari** [reply]
ya good point. mind opening issue on making extend simpler for workspace like project? i have sone idea but need an issue to track it

**Reply by jayvdb** [reply]
Created https://github.com/sagiegurari/cargo-make/issues/1175

**Comment by jayvdb** [comment]
Solved by https://github.com/sagiegurari/cargo-make/releases/tag/0.37.23
